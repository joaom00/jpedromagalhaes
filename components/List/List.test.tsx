import { render, screen } from '@testing-library/react'
import * as List from '.'

describe('<List />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <List.Root>
        <List.Item href="/item1" isActive>
          Item 1
        </List.Item>
        <List.Item href="/item2">Item 2</List.Item>
      </List.Root>
    )

    const link1 = screen.getByRole('link', { name: 'Item 1' })
    const link2 = screen.getByRole('link', { name: 'Item 2' })

    expect(link1).toBeInTheDocument()
    expect(link1).toHaveClass('bg-mauve5', 'dark:bg-mauveDark5')
    expect(link1).toHaveAttribute('href', '/item1')

    expect(link2).toBeInTheDocument()
    expect(link2).toHaveClass('hover:bg-mauve4', 'dark:hover:bg-mauveDark4')
    expect(link2).toHaveAttribute('href', '/item2')

    expect(container.firstChild).toMatchSnapshot()
  })
})
