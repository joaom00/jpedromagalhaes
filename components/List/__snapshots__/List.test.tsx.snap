// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render correctly 1`] = `
<ul
  class="space-y-3 px-3 py-4"
>
  <li>
    <a
      class="flex flex-col gap-2 rounded-md px-3 py-2 hover:bg-mauve4 dark:hover:bg-mauveDark4 bg-mauve5 dark:bg-mauveDark5"
      href="/item1"
    >
      Item 1
    </a>
  </li>
  <li>
    <a
      class="flex flex-col gap-2 rounded-md px-3 py-2 hover:bg-mauve4 dark:hover:bg-mauveDark4 hover:bg-mauve4 dark:hover:bg-mauveDark4"
      href="/item2"
    >
      Item 2
    </a>
  </li>
</ul>
`;
